hathor/event/event_manager.py

```diff
@@ -42,9 +42,63 @@
     HathorEvents.REORG_FINISHED,
     HathorEvents.TX_METADATA_CHANGED,
     HathorEvents.BLOCK_METADATA_CHANGED,
+    HathorEvents.CONSENSUS_TX_UPDATE,
+    HathorEvents.CONSENSUS_TX_REMOVED,
 ]
 
 
+def _todo(data: Dict[str, Any]) -> Dict[str, Any]:
+    raise NotImplementedError('TODO')
+
+
+def _empty(data: Dict[str, Any]) -> Dict[str, Any]:
+    return {}
+
+
+def _extract_tx(data: Dict[str, Any]) -> Dict[str, Any]:
+    tx = data['tx']
+    return {
+        'hash': tx.hash_hex,
+        # TODO: the rest
+    }
+
+
+def _extract_reorg(data: Dict[str, Any]) -> Dict[str, Any]:
+    return {
+        'reorg_size': data['new_best_height'] - data['old_best_height'],
+        'previous_best_block': data['old_best_block'].hash_hex,
+        'new_best_block': data['new_best_block'].hash_hex,
+    }
+
+
+_EVENT_EXTRACT_MAP: Dict[HathorEvents, Callable[[Dict[str, Any]], Dict[str, Any]]] = {
```

Comment:
Shouldn't it be `Dict[HathorEvents, Callable[[EventArguments, Dict[str, Any]]]`?

Reply:
That's a possible alternative, but the annotation matches the current implementation, which feeds a dict to the function, not the event.

Reply:
What's the benefit of passing the dict instead of the event itself? I mean, each extractor might access `event_args.__dict__` if they need.

Reply:
The only "benefit" is that I expect all extractors to just use the `data`, so they might as take the dict directly.

Reply:
You've already refactored this part. Thanks!